### PR TITLE
refactor: Refactor string input checks

### PR DESCRIPTION
**Pull request by yinggeh**
#### What does the PR do?
Move each individual check of string input in model backend repos to the bakend repo. 

#### Checklist
- [x] PR title reflects the change and is of format `<commit_type>: <Title>`
- [x] Changes are described in the pull request.
- [x] Related issues are referenced.
- [x] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [x] Verified copyright is correct on all changed files.
- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [x] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [x] refactor

#### Related PRs:
backend: https://github.com/triton-inference-server/backend/pull/101
tensorflow_backend: https://github.com/triton-inference-server/tensorflow_backend/pull/104
pytorch_backend: https://github.com/triton-inference-server/pytorch_backend/pull/136
onnxruntime_backend: https://github.com/triton-inference-server/onnxruntime_backend/pull/263
python_backend: https://github.com/triton-inference-server/python_backend/pull/370

#### Test plan:
- CI Pipeline ID:
17021222 and 17057045

#### Background
Initially just to add missing string input tensor checks to python_backend. Refactored because duplicate code from other model backends were found.

### CODE REVIEW

File: src/tensorflow.cc [hunk 1]
```diff
@@ -582,8 +582,11 @@ SetStringInputTensor(
       &contiguous_buffer, stream, &cuda_copy);
   if (err != nullptr) {
     RESPOND_AND_SET_NULL_IF_ERROR(response, err);
-    FillStringTensor(
```
[the remark this answers is not in the document]
Comment (tanmayv25) [reply src/tensorflow.cc:585]: Looks like this dates back to when the support for the string tensors was added to the tensorflow backend.
See [here](https://github.com/triton-inference-server/server/commit/c14e51af27c702ddc590b75a6d65225a701b25a1).

I am not sure at this time. A quick check could be removing these calls and introduce some errors in GetContiguousInputContent call.

File: src/tensorflow.cc [hunk 2]
```diff
@@ -582,8 +582,11 @@ SetStringInputTensor(
...
-        tensor, tensor_offset + element_idx, request_element_cnt - element_idx);
+    if (element_idx < request_element_cnt) {
```
Comment (tanmayv25) [inline src/tensorflow.cc:585]: `element_idx` will always be 0 here right?  Any reason for not treating it as 0 here and in L586?

We can then change the for loop to look like:

```
for (size_t element_idx=0; element_idx < str_list.size(); ++element_idx) {
```

Do you see any reason for not making this change?

File: src/tensorflow.cc [hunk 3]
```diff
@@ -595,68 +598,20 @@ SetStringInputTensor(
...
+    TRITONTF_TensorSetString(tensor, tensor_offset + element_idx, addr, len);
+  }
 
-    TRITONTF_TensorSetString(tensor, tensor_offset + element_idx, content, len);
-    content += len;
-    content_byte_size -= len;
-    element_idx++;
-  }
-
-  if ((*response != nullptr) && (element_idx != request_element_cnt)) {
```
Comment (tanmayv25) [inline src/tensorflow.cc:647]: `(element_idx != request_element_cnt)`

Don't we need this check? I think with the current changes it should look something like below:

```
request_element_cnt != str_list.size()
```

Comment (tanmayv25) [reply src/tensorflow.cc:647]: I see.. The check is part of ValidateStringBuffer.

File: src/tensorflow.cc (hunk 2 above)
Comment (yinggeh) [reply src/tensorflow.cc:585]: Agree. But `element_idx` needs to be outside of the for loop.

**Review by Tabrizian — DISMISSED** [review]

File: src/tensorflow.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (yinggeh) [reply src/tensorflow.cc:585]: Created a ticket for tracking (DLIS-7114).

**Review by tanmayv25 — APPROVED** [review]
